File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxi = 0;
        int n = height.length;

        int left = 0;
        int right = n - 1;

        while (left < right) {

            int width = right - left;
            int length = Math.min(height[left], height[right]);
            int area = length * width;

            if (area > maxi) {
                maxi = area;
            }

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxi;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxi = 0
        n = len(height)
        left = 0
        right = n-1
        while(left<right):
            width = right - left
            length = min(height[left],height[right])
            area = length*width
            if(area>maxi): maxi=area
            if(height[left]<height[right]):
                left+=1
            else:
                right-=1
        return maxi
